IbFlexReader/IbFlexReader/Reader.cs

```diff
@@ -2,92 +2,215 @@
+		/// <param name="xmlFile"></param>
+		/// <param name="options"></param>
+		/// <returns>An object with account activities</returns>
+		public Contracts.FlexQueryResponse GetByString(string xmlFile, Options options = null)
+		{
+			using (var stream = this.sb.GenerateStream(xmlFile))
+			{
+				var result = Deserializer.Deserialize<Xml.Contracts.FlexQueryResponse, Contracts.FlexQueryResponse>(stream, out var errorObjects);
+				result = result ?? new Contracts.FlexQueryResponse();
+				result.Errors = errorObjects;
+				Logic.ProcessStatement(result.FlexStatements?.FlexStatement, options);
+
+				return result;
+			}
+		}
 
         /// <summary>
         /// Reads a FlexQuery by API.
         /// </summary>
         /// <param name="token">token provided by IB</param>
         /// <param name="queryId">ID of the FlexQuery (not the name)</param>
         /// <param name="dumpFile">if set, the data returned from IB will be dumped to file</param>
+        /// <param name="retry">if GetStatement returns warning 1019 (... Please try again shortly), retry defines how many times it will be tried again</param>
         /// <returns></returns>
-        public async Task<FlexResult> GetByApi(string token, string queryId, string dumpFile = null)
-        {
-            using (var client = new HttpClient())
-            {
-                var uri = new Uri($"https://gdcdyn.interactivebrokers.com/Universal/servlet/FlexStatementService.SendRequest?t={token}&q={queryId}&v=3");
-
-                string referenceCode;
-                using (var sendRequestResult = await client.PostAsync(uri, null))
-                using (var stream = await sendRequestResult.Content.ReadAsStreamAsync())
-                {
-                    if (dumpFile != null) { Utils.DumpFileWriter.DumpSendRequest(stream, dumpFile); }
-
-                    var response = Deserializer.Deserialize<XmlFlexStatementResponse, FlexStatementResponse>(stream, out var errorObjects);
-
-                    if (response.Status != "Success")
-                    {
-                        return new FlexResult
-                        {
-                            ErrorCode = response.ErrorCode,
-                            ErrorMessage = response.ErrorMessage
-                        };
-                    }
-
-                    referenceCode = response.ReferenceCode.Value.ToString();
-                }
-
-
-                uri = new Uri($"https://gdcdyn.interactivebrokers.com/Universal/servlet/FlexStatementService.GetStatement?q={referenceCode}&t={token}&v=3");
-
-                using (var getStatementResult = await client.PostAsync(uri, null))
-                using (var stream = await getStatementResult.Content.ReadAsStreamAsync())
+        public async Task<FlexResult> GetByApi(string token, string queryId, string dumpFile = null, int? retry = null)
+		{
+			using (var client = new HttpClient())
+			{
+				FlexResult requestResult = await SendRequest(client, token, queryId, dumpFile);
```

Comment:
Can you try to use `await xxxxxx().ConfigureAwait(false)` everywhere and check if this does not break something? This will enable the library user to call `getByApi().Result` without deadlocking when called within the main GUI thread or ASP Controller action. I have made this wrong the first time. Sorry for that.

Reply:
Done